### ISSUE TITLE
0.56.x Discussion

### ISSUE THREAD

**Issue by grabbou** [react-native-community/react-native-releases#14]
Please report any major issues and commits to cherry-pick in this thread for discussion.

**Comment by iRoachie** [comment]
Could we get this in? https://github.com/facebook/react-native/commit/746945824a426a6f16555df0f12f92d836d5cc18

**Comment by grabbou** [comment]
It's not yet cut, so everything in master lands into it.

**Comment by grabbou** [comment]
Not really. I am probably going to cut 0.56 before Babel 7 change as it
still needs some fixes to get master green. We will do it for the 0.57 I
guess.

**Comment by Titozzz** [comment]
Oh well, I guess I'll wait one more then. Can't wait for this ! Babel 7 is finally coming. 🎉

**Comment by maraujop** [comment]
This was requested in 0.55.x discussion https://github.com/react-native-community/react-native-releases/issues/11#issuecomment-380121877 and I'm requesting it again for 0.56.x a fix for TextInputs in iOS https://github.com/facebook/react-native/pull/18278

**Comment by MarkOSullivan94** [comment]
Can we please get this sorted ASAP? https://github.com/facebook/react-native/issues/18095

Don't want to see a lot of RN projects unable to update their Android apps on the store in Nov 2018. Or new projects not being able to publish their apps in August 2018.

**Comment by grabbou** [comment]
This will get addressed at some point since Facebook and other companies
are also using React Native. Don't be worried. Unfortunately, this thread
is not the right place to discuss it.

But you have my word it's being worked on.

**Comment by MarkOSullivan94** [comment]
Also curious if there's any plans to include Gradle 3.x support in this release?

https://github.com/facebook/react-native/pull/17747

**Comment by MarkOSullivan94** [comment]
I don't need help upgrading to either, I'm using target version 27 and the latest version of gradle which comes with AS 3.2 canary 14.

Posting these issues just because there might be a lot of people who are on old versions which will need to be upgraded (target api a higher priority just because it'll stop RN from releasing updates / being publish in the not to distant future)

**Comment by vrgimael** [comment]
I looked up in many places and couldn't really find any info... what's the status on the 0.56 release? 

Thanks

**Comment by turnrye** [comment]
Work is taking place to get master back to a stable state. There are multiple efforts presently underway for this. I suspect that late next week, RC activities will pick up.

**Comment by grabbou** [comment]
Yup, currently Hector from Facebook (@hramos) is leading internal efforts on making the CI green again. We will use this thread to notify every one of the progress (if any).

I also hope that CI gets green soon and next RC gets released, but given that we are slowly approaching an end of this month, I'll be looking at the beginning of June as a potential time for next RC.

**Comment by rawrmaan** [comment]
Would love to know more about the RN Fabric efforts going into this release. I can't seem to find any information about Fabric, its advantages, roadmap, etc.

**Comment by sibelius** [comment]
Gradle 3 support would make build faster

**Comment by ntomallen** [comment]
Agreed. We switched to gradle 3+ and build tools 27+ without much issue ourselves, but would be nice to have it up-to-date by default.

**Comment by jsdario** [comment]
Is there a PR ready for that? I think gradle 3 is a nice-to-have, but probably should not be a priority for a release candidate, but rather something that can land before v0.57.

**Comment by MarkOSullivan94** [comment]
@jsdario I already posted a link to the PR for it: https://github.com/facebook/react-native/pull/17747

**Comment by MarkOSullivan94** [comment]
Could someone help me with this issue? I've made changes to the React Native Share API a week ago and I'm wanting to test my changes to ensure that Android can now share images now: https://github.com/facebook/react-native/issues/19358

**Comment by grabbou** [comment]
Please keep this thread related to commits that are already merged to React Native master. Meanwhile, if you need any additional help in the main React Native repo, feel free to ping me or Hector.

**Comment by kelset** [comment]
A couple of commits I'd like to see in this release:

* https://github.com/facebook/react-native/commit/41975f75d96ef4b606b4618461bf24d5db063b77 (fix for some weird GBoard interactions with textinput)
* https://github.com/facebook/react-native/commit/122b3791ede095345f44666691aa9ce5aa7f725a (blob memory leak)

**Comment by musicode** [comment]
When to release this version?

**Comment by kelset** [comment]
Quick update, @grabbou said that more or less today (https://twitter.com/grabbou/status/1005783171298885632) the new release will be cut.

To answer @musicode, yes technically master is stable.

**Comment by anandprabhu** [comment]
Thanks for the update @kelset i hope this also includes the fix for the sourcemap issue in the `metro` bundler by @rafeca is also included 

- https://github.com/facebook/metro/commit/c907c37a2096f1fc8979cdaa322fc2152b1944c7

**Comment by hramos** [comment]
### The 0.56 Release Candidate is now available

The new release has been cut. You may now use `v0.56.0-rc`. 

### Requesting backports

May I kindly request that any future conversation on this thread be limited to backport (cherry-pick) requests from commits that _are already on `master`_. 

An example of a good such request is a bug fix for a serious issue that has been merged into `master` but did not make the `0.56` RC cut. 

In other words, if you cannot point to [a particular commit on `master`](https://github.com/facebook/react-native/commits/master), then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

**Comment by MarkOSullivan94** [comment]
Out of interest was Gradle 3.x support removed?

**Comment by MarkOSullivan94** [comment]
Can we please have anything which didn't make the 0.56.0 release be added for 0.57.x or one of the 0.56.x releases (some clarification on the aims of what will be done and when would be perfect)

**Comment by kelset** [comment]
@MarkOSullivan94 as @hramos wrote, please

> May I kindly request that any future conversation on this thread be limited to backport (cherry-pick) requests from commits that are already on master.

Or I will lock this thread.

> Out of interest was Gradle 3.x support removed?

You can check the Changelog [here](https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md) and no, it wasn't removed. And it's not in the roadmap.

> Can we please have anything which didn't make the 0.56.0 release be added for 0.57.x or one of the 0.56.x releases (some clarification on the aims of what will be done and when would be perfect)

When 0.56.0 will be released, we'll create two new issues in this repo related to 0.57.x. To know which commits will land where, you can follow this issue and #15 (and the ones for 0.57.x once they are created).

**Comment by hramos** [comment]
I'd like to nominate https://github.com/facebook/react-native/commit/d0219a0301e59e8b0ef75dbd786318d4b4619f4c for cherry-picking onto 0.56. It fixes an issue with the "Dismiss All" button when YellowBox warnings are present which was introduced in `0.56.0-rc`.

**Comment by hramos** [comment]
Gradle 3.x support and requests related to Android for a future release are best suited for the Android roadmap umbrella issue in the main React Native repo: https://github.com/facebook/react-native/issues/19297

**Comment by hramos** [comment]
Here's a couple of issues that have been raised about `0.56.0-rc` so far:

- Wrong code for babel-preset-react-native@5.0.1 published 
 (https://github.com/facebook/react-native/issues/19666)
- Unexpected token import, v0.56.0-rc cliEntry.js error (https://github.com/facebook/react-native/issues/19669)

**Comment by hramos** [comment]
Cherry-picking https://github.com/facebook/react-native/commit/065c5b6590de18281a8c592a04240751c655c03c as requested by @dulmandakh in https://github.com/facebook/react-native/issues/19297#issuecomment-397060485

**Comment by dulmandakh** [comment]
From August 2018, Play Store will require to target SDK 26 for new applications and November for app updates.

So, I think that we must to target SDK 26 in August release to make sure applications developed with React Native be accepted in Play Store.

My plan was to compile July release with SDK 26 then target SDK 26 in August release (just in time) to make transition to SDK 26 smoother and to see if there are issues to address. Therefore, I would like to ask to land https://github.com/facebook/react-native/commit/065c5b6590de18281a8c592a04240751c655c03c in July release, so we have time to test, fix bugs and maybe land SDK 26 features like custom fonts, better emoji support etc.

**Comment by grabbou** [comment]
This commit should be part of next release, which will be RC next month and
become stable as of the beginning of August. Given that people usually do
not test RCs, I think it makes sense to cherry-pick this commit, esp. that
the branch has been like a few days ago.

Let's keep it on the list until the cherry-picks grow to a bigger number so
it's worth making next release.

**Comment by Ashoat** [comment]
I'd like to request a cherry-pick for https://github.com/facebook/react-native/commit/1ffb2b63be4c4af331fece0b4286e5c92b1e575d

It fixes a pretty bad iOS TextInput bug in master where fake backspace events get triggered.

**Comment by hramos** [comment]
@Ashoat done.

**Comment by mgenov** [comment]
Tried to switch to RN 0.56.0-rc.2 but getting "Unknown named module: 'NativeModules':

<img width="320" alt="screen shot 2018-06-16 at 18 22 51" src="https://user-images.githubusercontent.com/209223/41499985-82f45aea-7192-11e8-8740-cb02fefeb403.png">

package.json versions 
```jsx
    "react": "16.3.2",
    "react-dom": "16.3.2",
    "react-native": "^0.56.0-rc.2",
     ....
    "babel-preset-react-native": "5.0.2",
```


Build Tools & SDK

app/build.gradle
```groovy
compileSdkVersion 27
buildToolsVersion '27.0.3'
```

build.gradle
```groovy
classpath 'com.android.tools.build:gradle:3.1.2'
classpath 'com.google.gms:google-services:3.2.1'
```

Any idea what could cause the issue ?

**Comment by mgenov** [comment]
The problem was related with the remote-redux-devtools. It works as expected when the usage of remote-redux-devtools was removed.

**Comment by paolospag** [comment]
I'm really curious to try RN 0.56.0-rc.2 and I updated my RN 0.55.4 project via react-native-git-upgrade (previously I upgraded babel preset to 5.0.1 version).

The upgrade fails with some warnings and one error about parenthesis ("unexpected token...").

Is there a way to correctly upgrade to the newest version via react-native-git-upgrade?

**Comment by IljaDaderko** [comment]
@paolospag Try to update babel preset to 5.0.2

**Comment by eshikerya** [comment]
After upgrading via react-native-git-upgrade the app working in simulator, but crashing on init stage in production build on device. Removed all plugins, nothing works. Any clue?

**Comment by paolospag** [comment]
@IljaDaderko thank you for your answer!

I tried to update babel preset to 5.0.2, but I encountered the same error:
`git-upgrade info Install the new version
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.3 (node_modules\sane\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN jest-react-native@18.0.0 requires a peer of react-native@>=0.38.0 but none was installed.
npm WARN react-native-firebase@3.3.1 requires a peer of react-native@>= 0.48.0 but none was installed.
npm WARN react-native-material-menu@0.3.1 requires a peer of react-native@>= 0.54.0 but none was installed.
npm WARN react-native-splash-screen@3.0.6 requires a peer of react-native@>=0.47.0 but none was installed.
git-upgrade info Generate new version template
git-upgrade ERR! An error occurred during upgrade:
git-upgrade ERR! C:\Users\USER_NAME\Desktop\DEV\APP\node_modules\babel-preset-react-native\transforms\transform-symbol-member.js:44
            ),
            ^

SyntaxError: Unexpected token )
    at createScript (vm.js:56:10)
    at Object.runInThisContext (vm.js:97:10)
    at Module._compile (module.js:542:28)
    at Module._extensions..js (module.js:579:10)
    at Object.require.extensions.(anonymous function) [as .js] (C:\Users\USER_NAME\AppData\Roaming\npm\node_modules\react-native-git-upgrade\node_modules\babel-register\lib\node.js:152:7)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
git-upgrade ERR! Restore initial sources
M       package.json
Note: checking out 'project-snapshot'.`

Any idea about this?
I noticed that the new CHANGELOG says "_Node 8 is now the minimum required version_ [...]", but in my dev environment I'm still using Node.js 6.11.4... Could this be?

**Comment by kelset** [comment]
So, quick couple of new cherry picks requests (which could be optional):
- https://github.com/facebook/react-native/commit/12410f3e4ad9bb629b891dafeffdeeef9c114fb3 that would remove some warnings. Not totally sure about this since in the commit history on master looks like @hramos you [reverted it](https://github.com/facebook/react-native/commit/aaddbee29ef93c20b6f881def680ba30ce344a58) once then the bot merged again..?

- https://github.com/facebook/react-native/commit/a1302392577789faab79dad0cb39b147464e0e42 which should fix support for XCode 10 - again, not sure, since it depends on the feedback from this issue https://github.com/facebook/react-native/issues/19774

- https://github.com/facebook/react-native/commit/843a433e87b0ccaa64ab70d07e22bffbabad8045 fixes https://github.com/facebook/react-native/issues/19766

- https://github.com/facebook/react-native/commit/c749d951ada829c6f6fb76f35e68142e61054433 to update to React 16.4.1 as asked by @matthargett

**Comment by grabbou** [comment]
Unlocked this thread as by definition, this one is meant to be the place where the community can suggest cherry-picks and notify us of the issues with the release.

Please keep in mind that all discussions not related to releasing the new version should be held in the react-native repository.

**Comment by grabbou** [comment]
Not sure about the first commit - it's not something critical or needed for the release (like React) update.

I cherry-picked two last commits and scheduled next release. Will update the other issue when the status is green.

**Comment by EdwardDrapkin** [comment]
There are a couple PRs opened in this thread, can it get looked into for this release? https://github.com/facebook/react-native/issues/18272

**Comment by hramos** [comment]
@EdwardDrapkin please keep the thread on topic. We're only cherry-picking commits that are already on `master`.

**Comment by MarkOSullivan94** [comment]
@hramos is there an easy way to check what commits are in master but not latest 0.56.x release?

**Comment by stianjensen** [comment]
This shows all commits between the latest 0.56.x tag and master. You can find it by clicking on Releases, then the latest tag and "{x} commits to master since this tag"

https://github.com/facebook/react-native/compare/v0.56.0-rc.3...master

**Comment by vonovak** [comment]
can we get this small android permissions fix? https://github.com/facebook/react-native/commit/4e1abdd74dc4127a86d62e7750d01d39bb781c08

**Comment by kelset** [comment]
Since we are trying to support API 26 for Android in this release, I think we should cherry pick these two too:

* https://github.com/facebook/react-native/commit/b569154ae633d7f3c6b4f148b108f05aef742bf9

* https://github.com/facebook/react-native/commit/b5dc45420a0d3aa54d2d2075d7f14ff1835df78a

**Comment by MarkOSullivan94** [comment]
Couple of commits I'd like to see added

- https://github.com/facebook/react-native/commit/7b9b1559a7f6719c3c9ad8e894fcdd99ed109afe

- https://github.com/facebook/react-native/commit/75e49a0637eaa3bd3bb7e445648f084a42d9c8af

**Comment by grabbou** [comment]
Going to review the suggested cherry-picks today and will be releasing next release candidate tomorrow. That is most likely going to be one of the last RCs in preparations for the next month release candidate.

**Comment by eshikerya** [comment]
Does it mean the issue [#19827](https://github.com/facebook/react-native/issues/19827) will go to production?

**Comment by kelset** [comment]
@eshikerya 
>  one of the last RCs in preparations for the next month release

doesn't mean "will go in production". We hope that by the time 0.56.0 will be released a fix for that issue will have been found and merged.

**Comment by gastonmorixe** [comment]
+1 Can't wait for #19827 to be fixed. Blocked by it since depending on babel 7.

**Comment by hramos** [comment]
Thanks @grabbou for cherry picking all of these. `0.56` is scheduled to hit stable next Monday (first business day in July). After this last RC is cut, I think we should focus our attention on release-blocking issues such as #19827. Please cc me on any PR that attempts to fix #19827.

**Comment by grabbou** [comment]
Sure, that sounds like a good idea.

**Comment by gnprice** [comment]
> EXTRA: I think we should try to get this merged and cherry picked: facebook/react-native#19645

After some deeper debugging of the issue today, that's now a simple one-line bugfix, where the original was a revert of a previous commit that added a feature. So hopefully that now makes it easy to include!

**Comment by dulmandakh** [comment]
please cherry-pick https://github.com/facebook/react-native/commit/77a02c0d83dbfcd9a5397cf63e1ab2e6c94cfdde, so we can remove ReactFragmentActivity  and do some code cleanup in 0.57.

**Comment by hramos** [comment]
Note to whoever cherry picks these, when adding https://github.com/facebook/react-native/commit/5017b86b525e3ef6023f0f8a88e6fd1cf98024e0 to the changelog, please use the updated description (see my comment there).

**Comment by ikesyo** [comment]
I want https://github.com/facebook/react-native/commit/75a0273de21948b0b959263100f09111f738ec35 to be cherry-picked.

**Comment by futuun** [comment]
those 3 commits are fixing issues with Xcode 10, would be nice to have them:
https://github.com/facebook/react-native/commit/75a0273de21948b0b959263100f09111f738ec35 (for fishhook import)
https://github.com/facebook/react-native/commit/e131fffb37a545363daf11735a0243165b57f63f (for missing config.h in glog)
https://github.com/facebook/react-native/commit/a1302392577789faab79dad0cb39b147464e0e42 (for opening invalid simulator (or no simulator at all if you have only xcode beta))

**Comment by kelset** [comment]
Hopefully hours. I've tested tested with 3 different repos(debug/release, ios/Android), and the update worked on all these 3 scenarios :
- one of our own prod apps
- plain test app from 0.55.4 to 0.56.rc4 (and uploaded the repo here https://github.com/kelset/rn56-testupdate )
- another more complex test app with modules-export, .android/.ios files

and submitted a couple of PRs.
Now other devs in the core will review and approve/discuss and hopefully we can release 0.56.0.

Sadly it won't be as perfect as we'd like, but we need to keep moving forward IMHO.

**Comment by rozele** [comment]
Can we cherry-pick https://github.com/facebook/react-native/commit/54942746d4037e1153e14fcfc95e4edc772d296a to allow `react-native-windows` to work with 0.56?

**Comment by hramos** [comment]
All of the cherry-picks requested above have been added to the `0.56-stable` branch and should be released as part of `0.56.0-rc.5`. This is the final RC and no further commits will be cherry-picked unless a regression was introduced between rc.4 and rc.5 - we need to promote to stable at some point so we can get moving along.-

**Comment by dulmandakh** [comment]
Upgraded 2 projects from 0.55.4 to 0.56.0-rc.5 just fine. On iOS, got React Native version mismatch error and had to open XCode, then run Product -> Clean to fix it. On Android, resolved few conflicts due to changes in gradle files.

**Comment by angly-cat** [comment]
@dulmandakh How did you upgrade, btw? With `react-native-git-upgrade`? Or just by specifying newer version in `package.json`?

**Comment by dulmandakh** [comment]
@angly-cat **react-native-git-upgrade next**

**Comment by kelset** [comment]
### 0.56.0 has been released

Any future conversation on this thread be limited to backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into master but did not make the 0.56.0 cut. For new features that you would like to see implemented, refer to https://github.com/react-native-community/react-native-releases/issues/34

In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

**Comment by kelset** [comment]
For 0.56.1 we should aim (IMHO) to release as soon as this is fixed fully: https://github.com/facebook/react-native/issues/19953 (and https://github.com/facebook/metro/issues/181)

Currently it seems that a fix has been merged in master: https://github.com/facebook/react-native/commit/3ac86c366c91f8d62f0128057019b94a783b4249

but there has been a report that is actually still not fixed (sadly). So prob cherry picking that is a start but something else will be required.

**Comment by janicduplessis** [comment]
Would be nice to include b4b594c if we do a patch release, it fixes a regression with `TextInput.State.currentlyFocusedField`.

**Comment by hramos** [comment]
It looks like that commit can be applied cleanly on top of 0.56. If you have push access, you can try cherry-picking it onto 0.56-stable and running the bump-version script to tag a new patch release.

**Comment by grabbou** [comment]
Just did that before we make 0.57 stable (for some reason it doesn't work properly when we release a patch to non-latest release).

Anyway, impressed we didn't have much to cherry-pick and release since then.

**Comment by grabbou** [comment]
Closing as `0.57.0` has just hit npm. No future cherry-picks will be accepted unless a security/critical issue arises in the meantime.
